src/HPolyhedron.jl

```diff
@@ -340,6 +345,108 @@ function tohrep(P::HPoly{N}) where {N}
     return P
 end
 
+"""
+    remove_redundant_constraints(P::PT;
+                                 backend=GLPKSolverLP()) where {N, HPoly{N}}
+
+Given a polyhedron in H-representation, return a new polyhedron with no reundant
+constraints.
+
+## Input
+
+- `P`       -- polyhedron
+- `backend` -- (optional, default: `GLPKSolverLP`) the numeric LP solver backend
+
+## Output
+
+The polyhedron obtained by removing the redundant constraints in `P`.
+
+## Algorithm
+
+See `remove_redundant_constraints!`. 
+"""
+function remove_redundant_constraints(P::PT;
+                                      backend=GLPKSolverLP()) where {N, PT<:HPoly{N}}
+    return remove_redundant_constraints!(copy(P), backend=backend)
+end
+
+"""
+    remove_redundant_constraints!(P::PT;
+                                  backend=GLPKSolverLP()) where {N, HPoly{N}}
+
+Remove the reundant constraints in a polyhedron in H-representation; the polyhedron
+is updated inplace.
+
+## Input
+
+- `P`       -- polyhedron
+- `backend` -- (optional, default: `GLPKSolverLP`) the numeric LP solver backend
+
+## Output
+
+The polyhedron obtained by removing the redundant constraints in `P`.
+
+## Algorithm
+
+If the polyhedron `P` has `m` constraints and its dimension is `n`,
+this function checks one by one if each of the `m` constraints is
+implied by the `m-1` others. To check if the `k`-th constraint
```

Comment:
```suggestion
implied by the remaining ones. To check if the `k`-th constraint
```

Reply:
Reason for change: after removing one constraint, there are only `m-2` remaining, etc.